prime-router/metadata/hl7_mapping/common/observation-result.yml

```
@@ -165,30 +165,30 @@ elements:
     hl7Spec: [ '%{hl7OBXField}-19' ]
 
   - name: performing-organization-name-pracrole
-    condition: '%observation.performer.resolve() is PractitionerRole'
-    resource: '%resource.performer.resolve().organization.resolve()'
+    condition: '%observation.performer.resolve().ofType(PractitionerRole).exists() and %observation.performer.resolve().ofType(PractitionerRole).organization.exists()'
+    resource: '%resource.performer.resolve().ofType(PractitionerRole).organization.resolve()'
     schema: ./datatype/xon-organization
     constants:
       hl7OrgField: '%{hl7OBXField}-23'
 
   - name: performing-organization-name-non-pracrole
-    condition: '%observation.performer.resolve().is(PractitionerRole).not()'
-    resource: '%resource.performer.resolve()'
+    condition: '%observation.performer.resolve().ofType(PractitionerRole).exists().not() or %observation.performer.resolve().ofType(PractitionerRole).organization.exists().not()'
+    resource: '%resource.performer.resolve().ofType(Organization)'
     schema: ./datatype/xon-organization
     constants:
       hl7OrgField: '%{hl7OBXField}-23'
 
   - name: performing-organization-address-pracrole
-    condition: '%observation.performer.resolve() is PractitionerRole'
+    condition: '%observation.performer.resolve().ofType(PractitionerRole).exists() and %observation.performer.resolve().ofType(PractitionerRole).organization.exists()'
     resource: '%resource.performer.resolve().organization.resolve().address'
```

Guessing you meant to change this too?
```suggestion
    resource: '%resource.performer.resolve().ofType(PractitionerRole).organization.resolve().address'
```